interp/run.go

```diff
@@ -2034,7 +2034,7 @@ func doCompositeBinStruct(n *node, hasType bool) {
 			}
 		} else {
 			fieldIndex[i] = []int{i}
-			if c.typ.cat == funcT {
+			if c.typ.cat == funcT && len(c.child) > 1 {
```

Comment:
I do not understand this change.
I understand we are in the case of a literal composite, and one of its fields is defined without a key, just the value. and the value is a function. Could you give me an example of a case where
 `c.typ.cat == funcT && len(c.child) == 1`
and one where
 `c.typ.cat == funcT && len(c.child) == 2`
so i can understand the difference?

Reply:
This may happen when initializing a runtime defined struct with a function field (`typ.Field(i).Type == reflect.Func`) with an interpreter defined stuct with a function field already being a reflect.Value from runtime.

I met this case when interpreting (inception mode) `cmd/yaegi/test.go` at line 123:
```go
benchmarks = append(benchmarks, testing.InternalBenchmark{name, fun})
```
where `fun` is provided through a type switch assign, thus already converted to the runtime type.

Reply:
something's still unclear to me, but I'll clarify with you later. No need to block the PR for that.